plugins/stackstorm/src/api/StackStormClient.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Action, Execution, Pack, StackStormApi } from './types';
+import { DiscoveryApi, IdentityApi } from '@backstage/core-plugin-api';
+
+export class StackStormClient implements StackStormApi {
+  private readonly discoveryApi: DiscoveryApi;
+  private readonly identityApi: IdentityApi;
+
+  constructor({
+    discoveryApi,
+    identityApi,
+  }: {
+    discoveryApi: DiscoveryApi;
+    identityApi: IdentityApi;
```

Comment:
I recommend pass in `fetchApiRef` instead of the identity API; it is basically `fetch` but instrumented and overridable, and it injects the backstage auth token on your behalf without you having to think about it.